lib/LedgerSMB/Scripts/setup.pm

```diff
@@ -1436,9 +1436,11 @@ Force work.  Forgets unmatching tests, applies a curing statement and move on.
 sub force{
     my ($request) = @_;
     my $database = _init_db($request);
+    my $dbinfo = $database->get_info();
 
-    my %test = map { $_->name => $_ } LedgerSMB::Upgrade_Tests->get_tests();
-    my $force_queries = $test{$request->{check}}->{force_queries};
+    my %checks = map { $_->name => $_ } @{$info_applicable_for_upgrade{$dbinfo}};
+    my $check = $checks{$request->{check}};
```

Comment:
This statement and the one above can be done with `first()` from `List::Util` again.